packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/index.js

```diff
@@ -0,0 +1,1346 @@
+        actionType === 'edit'
+      ) {
+        const data = get(allDataSchema, pathToSchema, {});
+
+        dispatch({
+          type: 'SET_DATA_TO_EDIT',
+          data: {
+            name: data.schema.name,
+            category: data.category,
+            icon: data.schema.icon,
+            collectionName: data.schema.collectionName,
+          },
+        });
+      }
+
+      // Special case for the dynamic zone
+      if (
+        modalType === 'addComponentToDynamicZone' &&
+        state.modalType !== 'addComponentToDynamicZone' &&
+        actionType === 'edit'
+      ) {
+        const attributeToEditNotFormatted = get(
+          allDataSchema,
+          [...pathToSchema, 'schema', 'attributes', dynamicZoneTarget],
+          {}
+        );
+        const attributeToEdit = {
+          ...attributeToEditNotFormatted,
+          name: dynamicZoneTarget,
+          // createComponent: true,
+          createComponent: false,
+          componentToCreate: { type: 'component' },
+        };
+
+        dispatch({
+          type: 'SET_DYNAMIC_ZONE_DATA_SCHEMA',
+          attributeToEdit,
+        });
+      }
+
+      // Set the predefined data structure to create an attribute
+      if (
+        attributeType &&
+        attributeType !== 'null' &&
+        // This condition is added to prevent the reducer state to be cleared when navigating from the base tab to tha advanced one
+        state.modalType !== 'attribute'
+      ) {
+        const attributeToEditNotFormatted = get(
+          allDataSchema,
+          [...pathToSchema, 'schema', 'attributes', attributeName],
+          {}
+        );
+        const attributeToEdit = {
+          ...attributeToEditNotFormatted,
+          name: attributeName,
+        };
+
+        // We need to set the repeatable key to false when editing a component
+        // The API doesn't send this info
+        if (attributeType === 'component' && actionType === 'edit') {
+          if (!attributeToEdit.repeatable) {
+            set(attributeToEdit, 'repeatable', false);
+          }
+        }
+
+        if (
+          attributeType === 'relation' &&
+          !has(attributeToEdit, ['targetAttribute'])
+        ) {
+          set(attributeToEdit, ['targetAttribute'], '-');
+        }
+
+        dispatch({
+          type: 'SET_ATTRIBUTE_DATA_SCHEMA',
+          attributeType,
+          nameToSetForRelation: get(
+            sortedContentTypesList,
+            ['0', 'title'],
+            'error'
+          ),
+          targetUid: get(sortedContentTypesList, ['0', 'uid'], 'error'),
+          isEditing: actionType === 'edit',
+          modifiedDataToSetForEditing: attributeToEdit,
+          step,
+        });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search]);
+
+  const form = get(forms, [state.modalType, 'form', state.settingType], () => ({
+    items: [],
+  }));
+
+  // TODO improve icon logic
```

Comment:
The icon type should be store in the state so it's easier to modify the stepper.
This needs to be added in another PR